ENIGMAsystem/SHELL/Graphics_Systems/Direct3D9/DX9surface.cpp

```diff
@@ -41,6 +41,14 @@ namespace enigma {
 vector<Surface*> Surfaces(0);
 D3DCOLOR get_currentcolor();
 
+void surface_copy_to_systemmem(IDirect3DSurface9 **src, IDirect3DSurface9 **dest) {
+  D3DSURFACE_DESC desc;
+  (*src)->GetDesc(&desc);
+
+  d3dmgr->device->CreateOffscreenPlainSurface(desc.Width, desc.Height, desc.Format, D3DPOOL_SYSTEMMEM, dest, NULL);
+  d3dmgr->device->GetRenderTargetData(*src, *dest);
```

Comment:
Is there not an overload of this that allows retrieving a smaller window? Doing this for every call to `surface_getpixel` is INSANE.

Reply:
I don't think so, and that may be why this was so slow in older GM. The surface in VRAM is simply not lockable, period. I think OpenGL gets around this by caching the RAM copy... and flushing it at some point.

Reply:
Ok, I am going to ignore this for now, yes it's obviously slow, but that can be addressed later, for now, it at least works.

Reply:
Here: 8551c9c76074f719f8ff7a31df140f09aeee430b

I left a TODO comment so it doesn't get forgotten about.